File: withChris/0916/py/solution.py
def ArrayChallenge_O_lgN(arr):
    if len(arr) >= 3:
        mid_idx = len(arr) // 2
        if (arr[mid_idx + 1] - arr[mid_idx]) * (arr[mid_idx] - arr[mid_idx - 1]) < 0:
            return mid_idx
        else:
            ArrayChallenge_O_lgN(arr[0:mid_idx])
            ArrayChallenge_O_lgN(arr[mid_idx:len(arr)])

    return -1


print(ArrayChallenge_O_lgN([-4, -2, 9, 10]))
print(ArrayChallenge_O_lgN([5, 4, 3, 2, 10, 11]))

File: withChris/0916/js/solution.js
var ArrayChallenge_O_N = (arr) => {

    for (let idx = 2; idx < arr.length; idx++) {
        if ((arr[idx] - arr[idx - 1]) * (arr[idx - 1] - arr[idx - 2]) < 0) {
            return idx - 1;
        }
    }

    return -1;
};

var ArrayChallenge_DC_O_N = (arr) => {
    if (arr.length >= 3) {
        var midIdx = Math.floor(arr.length / 2);
        if ((arr[midIdx + 1] - arr[midIdx]) * (arr[midIdx] - arr[midIdx - 1]) < 0) {
            return midIdx;
        }
        else {
            ArrayChallenge_DC_O_N(arr.slice(0, midIdx));
            ArrayChallenge_DC_O_N(arr.slice(midIdx, arr.length));
        }
    }

    return -1
}

var findPeak = (arr) => {
    let left = 0;
    let right = arr.length - 1;

    while(){
        let mid = math.floor((right - left) / 2)
    }
}

console.log(ArrayChallenge_O_N([-4, -2, 9, 10]))
console.log(ArrayChallenge_O_N([5, 4, 3, 2, 10, 11]))

console.log(ArrayChallenge_DC_O_N([-4, -2, 9, 10]))
console.log(ArrayChallenge_DC_O_N([5, 4, 3, 2, 10, 11]))